src/args.rs

```diff
@@ -76,36 +76,43 @@ fn parse_interface(src: &str) -> Result<IpAddr, std::net::AddrParseError> {
 }
 
 /// Checks wether the auth string is valid, i.e. it follows the syntax username:password
-fn parse_auth(src: &str) -> Result<(String, String), String> {
-    let mut split = src.splitn(2, ':');
+fn parse_auth(src: &str) -> Result<auth::RequiredAuth, String> {
+    let mut split = src.splitn(3, ':');
+    let errmsg = "Invalid credentials string, expected format is username:password".to_owned();
 
     let username = match split.next() {
         Some(username) => username,
-        None => {
-            return Err(
-                "Invalid credentials string, expected format is username:password".to_owned(),
-            )
-        }
+        None => return Err(errmsg),
     };
 
-    let password = match split.next() {
+    let second_part = match split.next() {
         // This allows empty passwords, as the spec does not forbid it
         Some(password) => password,
-        None => {
-            return Err(
-                "Invalid credentials string, expected format is username:password".to_owned(),
-            )
-        }
+        None => return Err(errmsg),
     };
 
-    // To make it Windows-compatible, the password needs to be shorter than 255 characters.
-    // After 255 characters, Windows will truncate the value.
-    // As for the username, the spec does not mention a limit in length
-    if password.len() > 255 {
-        return Err("Password length cannot exceed 255 characters".to_owned());
-    }
+    let password = match split.next() {
+        Some(hash) => match second_part {
+            "sha256" => auth::RequiredAuthPassword::Sha256(hash.to_owned()),
+            "sha512" => auth::RequiredAuthPassword::Sha512(hash.to_owned()),
+            _ => return Err("Invalid hash method, valid methods is sha256".to_owned())
```

Comment:
You also accept sha512 but this error doesn't reflect that.

Reply:
My bad.